sway-lsp/src/error.rs

```diff
@@ -33,6 +33,10 @@ pub enum DocumentError {
     ManifestsLockPathFailed { dir: String },
     #[error("Document is already stored at {:?}", path)]
     DocumentAlreadyStored { path: String },
+    #[error("File wasn't able to be created: {:?}", err)]
+    UnableToCreateFile { err: String },
+    #[error("Unable to write string to file: {:?}", err)]
+    UnableToWriteFile { err: String },
```

Comment:
```suggestion
    UnableToWriteFile { err: io::Error },
```
Might be better to propagate the original error (which still impls `Debug` and `Display`) for these, but this could be done in a future PR as it looks like many of the other error variants are also currently "stringly typed".

Reply:
Nice, that's a good idea. Just made the suggested change.

Reply:
Actually, I just reverted this change as we need `Eq` and `PartialEq` implemented for our Error types so we can compare them in the LSP tests.